src/ui/ButtonView.js

```diff
@@ -0,0 +1,197 @@
+import ui.ImageView as ImageView;
+import ui.ImageScaleView as ImageScaleView;
+import lib.PubSub as PubSub;
+
+var uid = 0;
+var uidInput = -1;
+
+exports = Class(ImageScaleView, function(supr) {
+
+  this.init = function(opts) {
+    supr(this, 'init', [opts]);
+
+    if (!opts.pressedImage) {
+      opts.pressedImage = opts.image;
+    }
+
+    if (!opts.disabledImage) {
+      opts.disabledImage = opts.image;
+    }
+
+    this.normalImage = opts.image;
+    this.pressedImage = opts.pressedImage;
+    this.disabledImage = opts.disabledImage;
+
+    this.toggle = opts.toggle || false;
+
+    this._btnID = uid++;
+    this._pubsub = new PubSub();
```

Comment:
View's inherit from PubSub, aka Emitter, doesn't seem like adding an additional instance of PubSub is necessary